Feature/hw 1 1/impl

KonstantinPimbursky opened the pull request:
Домашнее задание к занятию 1.1 Структура проекта и жизненный цикл приложения iOS

KonstantinPimbursky committed:
Удален SceneDelegate
Перенесены методы из SceneDelegate в AppDelegate
Настроен window.rootViewController

KonstantinPimbursky committed:
Инициализация приложения с UITabBarController

KonstantinPimbursky committed:
Заменил SF Symbols и SystemColor
Добавил новый target к проекту
Добавил scheme сборки
Прописал 'print' в каждом методе жизненного цикла приложения

KonstantinPimbursky committed:
Время работы в бэкграунде записано в комментариях в методе applicationDidEnterBackground

TrueMax commented on Navigation/AppDelegate.swift:
```diff
@@ -11,25 +11,33 @@ import UIKit
+    
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        // Override point for customization after application launch.
+        self.window = UIWindow(frame: UIScreen.main.bounds)
```
лишние `self.` отсюда и ниже

TrueMax commented on Navigation/AppDelegate.swift:
```diff
@@ -11,25 +11,33 @@ import UIKit
+        self.window = UIWindow(frame: UIScreen.main.bounds)
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        let vc = storyboard.instantiateViewController(withIdentifier: "TabBarController")
+        self.window!.rootViewController = vc
```
optional chaining достаточно, не надо `force unwrap` применять к  свойству `window` 
пример: `window?.rootViewController = ...`

TrueMax commented on Navigation/AppDelegate.swift:
```diff
@@ -11,25 +11,33 @@ import UIKit
+    
+    func applicationDidEnterBackground(_ application: UIApplication) {
+        print(type(of: self), #function)
+        //По умолчанию в бэкграунде работа запрещена. Время работы в бэкграунде меньше секунды.
```
для выполнения задания 2 есть ссылка на материал, там 2 абзаца внизу нужно выполнить, готовая инструкция по запуску `backgroundTask`

KonstantinPimbursky committed:
В AppDelegate убраны лишние self.
У свойства window убран force unwrap и заменен на optional chaining

KonstantinPimbursky committed:
Добавлено backgroundTask в FeedViewController
Произведен замер времени работы приложения в background
Это время записано в комментарии в методе applicationDidEnterBackground
